woocommerce/google-listings-and-ads
# Free Listings + Paid Ads: Add the billing setup card and the paid ads completion

### Changes proposed in this Pull Request:

This PR implements the **📌 Billing Ads card** in #1610.

- Change to open a resizable popup window for setting up the billing of Google Ads account.
- Add an alternative link for opening the billing setup page
- Add a card for notifying the successfully added billing method.
- Complete the onboarding with paid ads setup.

### Screenshots:

https://user-images.githubusercontent.com/17420811/190395908-89480b8d-896b-4ea3-ae80-4b0970ed5042.mp4

### Detailed test instructions:

1. Preparation:
   - Disconnect Google Ads account
   - Delete some GLA settings to enter the onboarding flow
      ```sql
      DELETE FROM wp_options WHERE option_name = 'gla_mc_setup_completed_at';
      ```
2. Go to the last step of onboarding flow.
3. Connect to or Create a Google Ads account 
4. Check if the UI is similar to the visual design on Figma.
5. Click on the "click here instead" link should open Google's ads billing setup page on a new browser tab.
6. Click on the "Set up billing" button should open Google's ads billing setup page with a popup window.
   - The window should be resizable.
   - The window should vertically and horizontally align to the center of the current page.
7. Finish the billing setup and go back to the onboarding flow. Check if the statue is changed to a success notice after a while.
   - To skip adding real billing info, add the code snippet to the bottom of google-listings-and-ads.php on your local site:
      ```php
      add_filter(
      	'woocommerce_gla_ads_billing_setup_status',
      	function( $status ) {
      		return 'approved';
      	}
      );
      ```
8. Enter a campaign budget and finish the onboarding with the "Complete setup" button.
9. After redirecting to the Product Feed page, navigate to the Dashboard page to see whether the just-submitted paid campaign has been created.

💡 Since the states of Google Ads connection and its billing setup are standalone and may be on a middle status, it would be nice if try to refresh at any time to see if the UI could resume from a middle status. Please note that other runtime states such as the state of "show paid ads setup sections" will be handled in a separate PR later.

### Changelog entry
